gnucash/gnome-utils/dialog-utils.c

```diff
@@ -876,3 +879,187 @@ gnc_get_negative_color (void)
+        for (i = 0; !scm_is_null (rpt_guids); i++)
+        {
+            gchar *guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
+            gchar *name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name,
+                                                  SCM_CAR(rpt_guids), SCM_BOOL_F));
+
+            gtk_list_store_append (store, &iter);
+            gtk_list_store_set (store, &iter,
+                                COL_INV_NAME, name,
+                                COL_INV_GUID, guid_str,
+                                -1);
+            g_free (name);
+            g_free (guid_str);
+
+            rpt_guids = SCM_CDR(rpt_guids);
+        }
+    }
+}
+
+static gboolean
+select_default (GtkWidget *combo, const gchar *default_guid)
+{
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(combo));
+    GtkTreeIter iter;
+    gboolean found = FALSE;
+    gboolean valid_iter = gtk_tree_model_get_iter_first (model, &iter);
+
+    while (valid_iter)
+    {
+        gchar *guid;
+        gtk_tree_model_get (model, &iter, COL_INV_GUID, &guid, -1);
+
+        if (g_strcmp0 (default_guid, guid) == 0)
+        {
+            gtk_combo_box_set_active_iter (GTK_COMBO_BOX(combo), &iter);
+            g_free (guid);
+            found = TRUE;
+            break;
+        }
+        g_free (guid);
+        valid_iter = gtk_tree_model_iter_next (model, &iter);
+    }
+    return found;
+}
+
+static void
+combo_changed_cb (GtkComboBox *widget, gpointer user_data)
+{
+    GtkTreeIter iter;
+    QofBook *book = gnc_get_current_book();
+
+    if (gtk_combo_box_get_active_iter (widget, &iter))
+    {
+        GtkTreeModel *model = gtk_combo_box_get_model (widget);
+        gchar *name, *guid;
+        gtk_tree_model_get (model, &iter, COL_INV_NAME, &name,
+                                          COL_INV_GUID, &guid, -1);
+
+        if (g_strcmp0 (name, _("Missing Invoice Report")) == 0)
+            gtk_widget_show (user_data);
+        else
+            gtk_widget_hide (user_data);
+
+        gncInvoiceSetDefaultReport (book, guid);
+
+        gtk_widget_queue_resize (GTK_WIDGET(widget));
+        g_free (name);
+        g_free (guid);
+    }
+}
+
+void
+gnc_default_print_report_list (GtkWidget *combo, GtkWidget *warning)
+{
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(combo));
+    const gchar *default_guid = gnc_get_default_invoice_print_report ();
+    gboolean have_default = FALSE;
+
+    update_invoice_list (GTK_LIST_STORE(model));
+    have_default = select_default (combo, default_guid);
+
+    if (!have_default)
+    {
+        GtkTreeIter iter;
+        gchar *tool_tip = g_strdup_printf (_("The Invoice Report with Guid '%s' is missing"),
+                                           default_guid);
+
+        gtk_list_store_prepend (GTK_LIST_STORE(model), &iter);
+        gtk_list_store_set (GTK_LIST_STORE(model), &iter,
+                            COL_INV_NAME, _("Missing Invoice Report"),
```

Comment:
Can you add a translator comment where it appears, resp. why it is capitalized?

Reply:
I have decided to save the report name also so this has changed.